matrix-sdk/src/main/java/org/matrix/androidsdk/features/integrationmanager/IntegrationManager.kt

```diff
@@ -106,17 +113,66 @@ class IntegrationManager(val mxSession: MXSession, val context: Context) {
                     it[stateEventId] = allowed
                 }
 
+        val updatedMap = (
+                mxSession.dataHandler.store
+                        ?.getAccountDataElement(AccountDataElement.ACCOUNT_DATA_TYPE_ALLOWED_WIDGETS)
+                        ?.content ?: HashMap()
+                ).apply {
+            set("widgets", accountDataContent)
+        }
+
+        //optimistic update
+        widgetPermissions =
+                // Remove existing perm for current widget if any, then add updated state
+                widgetPermissions.filterNot { it.stateEventId == stateEventId } + listOf(WidgetPermission(stateEventId, allowed))
+        notifyListeners()
+
+        mxSession.accountDataRestClient.setAccountData(
+                mxSession.myUserId,
+                AccountDataElement.ACCOUNT_DATA_TYPE_ALLOWED_WIDGETS,
+                updatedMap,
+                object : SimpleApiCallback<Void>(callback) {
+                    override fun onSuccess(info: Void?) {
+                        callback?.onSuccess(null)
+                    }
+                })
+    }
+
+    fun seVideoConfDomainAllowed(domain: String, allowed: Boolean, callback: ApiCallback<Void?>?) {
+        val accountDataContent = videoConfDomainPermissions
+                //transform list to a map
+                .groupBy {
+                    it.domain
+                }
+                .mapValues {
+                    it.value.first().allowed
+                }
+                .toMutableMap()
+                .also {
+                    it[domain] = allowed
+                }
+
+        //Avoid to override delete unknwon keys
+        val updatedMap = (
+                mxSession.dataHandler.store
+                        ?.getAccountDataElement(AccountDataElement.ACCOUNT_DATA_TYPE_ALLOWED_WIDGETS)
+                        ?.content ?: HashMap()
+                ).apply {
+            set("jitsi", accountDataContent)
```

Comment:
"jitsi" should be pass in parameter "nativeWidgetName" or something like that
FTR nativeWidgetDomainPermission as map (key example: jitsi)